src/Raven.Server/Documents/DocumentsStorage.cs

```diff
@@ -1323,6 +1323,43 @@ public IEnumerable<ReplicationBatchItem> GetTombstonesFrom(DocumentsOperationCon
             }
         }
 
+        public long TombstonesCountForCollection(DocumentsOperationContext context, string collection)
+        {
+            string tableName;
+
+            if (collection == AttachmentsStorage.AttachmentsTombstones ||
+                collection == RevisionsStorage.RevisionsTombstones)
+            {
+                tableName = collection;
+            }
+            else
+            {
+                CollectionName collectionName;
+                try
+                {
+                    collectionName = GetCollection(collection, throwIfDoesNotExist: false);
+                }
+                catch
+                {
+                    return 0;
+                }
+
+                tableName = collectionName.GetTableName(CollectionTableType.Tombstones);
+            }
+
+            var table = context.Transaction.InnerTransaction.OpenTable(TombstonesSchema, tableName);
+            if (table == null)
+                return 0;
+
+            long count = 0;
+            foreach (var result in table.SeekForwardFrom(TombstonesSchema.FixedSizeIndexes[CollectionEtagsSlice], 0, 0))
+            {
+                count++;
```

Comment:
What if we have 1M tombstones?
Let's return here: `{ Count: count, Accuracy: Accurate | Apporoximate }`
If we have more than 10K tombstones, in the warning message we'll say `Collection 'collection' has more than 10K tombstone`

Reply:
done.